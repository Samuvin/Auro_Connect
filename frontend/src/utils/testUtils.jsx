import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 