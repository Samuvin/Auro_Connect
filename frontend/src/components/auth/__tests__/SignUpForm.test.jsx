import { screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUpForm from '../SignUpForm';
import { renderWithProviders } from '../../../utils/testUtils';

// Mock the dependencies
jest.mock('../../../lib/axios');

describe('SignUpForm', () => {
  test('renders form correctly', () => {
    renderWithProviders(<SignUpForm />);
    
    expect(screen.getByPlaceholderText('Full name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Username')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password (6+ characters)')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /agree & join/i })).toBeInTheDocument();
  });

  test('handles form submission', async () => {
    const user = userEvent.setup();
    renderWithProviders(<SignUpForm />);
    
    const nameInput = screen.getByPlaceholderText('Full name');
    const usernameInput = screen.getByPlaceholderText('Username');
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password (6+ characters)');
    const submitButton = screen.getByRole('button', { name: /agree & join/i });
    
    await user.type(nameInput, 'John Doe');
    await user.type(usernameInput, 'johndoe');
    await user.type(emailInput, 'john@example.com');
    await user.type(passwordInput, 'password123');
    
    fireEvent.click(submitButton);
    
    // Form should handle submission
    expect(submitButton).toBeInTheDocument();
  });

  test('validates required fields', () => {
    renderWithProviders(<SignUpForm />);
    
    const nameInput = screen.getByPlaceholderText('Full name');
    const usernameInput = screen.getByPlaceholderText('Username');
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password (6+ characters)');
    
    expect(nameInput).toBeRequired();
    expect(usernameInput).toBeRequired();
    expect(emailInput).toBeRequired();
    expect(passwordInput).toBeRequired();
  });

  test('email field has correct type', () => {
    renderWithProviders(<SignUpForm />);
    
    const emailInput = screen.getByPlaceholderText('Email');
    expect(emailInput).toHaveAttribute('type', 'email');
  });

  test('password field has correct type', () => {
    renderWithProviders(<SignUpForm />);
    
    const passwordInput = screen.getByPlaceholderText('Password (6+ characters)');
    expect(passwordInput).toHaveAttribute('type', 'password');
  });

  // Snapshot Tests
  test('matches snapshot for signup form default state', () => {
    const { container } = renderWithProviders(<SignUpForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for signup form with filled fields', async () => {
    const user = userEvent.setup();
    const { container } = renderWithProviders(<SignUpForm />);
    
    const nameInput = screen.getByPlaceholderText('Full name');
    const usernameInput = screen.getByPlaceholderText('Username');
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password (6+ characters)');
    
    await user.type(nameInput, 'John Doe');
    await user.type(usernameInput, 'johndoe');
    await user.type(emailInput, 'john@example.com');
    await user.type(passwordInput, 'password123');
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for signup form focused state', async () => {
    const user = userEvent.setup();
    const { container } = renderWithProviders(<SignUpForm />);
    
    const nameInput = screen.getByPlaceholderText('Full name');
    await user.click(nameInput);
    
    expect(container.firstChild).toMatchSnapshot();
  });
}); 