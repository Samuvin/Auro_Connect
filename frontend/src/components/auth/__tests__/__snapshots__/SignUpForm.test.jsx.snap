// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignUpForm matches snapshot for signup form default state 1`] = `
<form
  class="flex flex-col gap-4"
>
  <input
    class="input input-bordered w-full"
    placeholder="Full name"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Email"
    required=""
    type="email"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password (6+ characters)"
    required=""
    type="password"
    value=""
  />
  <button
    class="btn btn-primary w-full text-white"
    type="submit"
  >
    Agree & Join
  </button>
</form>
`;

exports[`SignUpForm matches snapshot for signup form focused state 1`] = `
<form
  class="flex flex-col gap-4"
>
  <input
    class="input input-bordered w-full"
    placeholder="Full name"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Email"
    required=""
    type="email"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password (6+ characters)"
    required=""
    type="password"
    value=""
  />
  <button
    class="btn btn-primary w-full text-white"
    type="submit"
  >
    Agree & Join
  </button>
</form>
`;

exports[`SignUpForm matches snapshot for signup form with filled fields 1`] = `
<form
  class="flex flex-col gap-4"
>
  <input
    class="input input-bordered w-full"
    placeholder="Full name"
    required=""
    type="text"
    value="John Doe"
  />
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value="johndoe"
  />
  <input
    class="input input-bordered w-full"
    placeholder="Email"
    required=""
    type="email"
    value="john@example.com"
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password (6+ characters)"
    required=""
    type="password"
    value="password123"
  />
  <button
    class="btn btn-primary w-full text-white"
    type="submit"
  >
    Agree & Join
  </button>
</form>
`;
